src/hooks/useMaxSiblingWidth.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import Path from '../@types/Path'
+import editingValueStore from '../stores/editingValue'
+import parentOf from '../util/parentOf'
+
+/**
+ * Helper function to collect nodes with the specified `data-parent-path`.
+ * If no thought node is found, it continues searching with `data-grand-parent-path`.
+ */
+const collectNodes = (path: string): HTMLElement[] => {
+  /** Collect nodes by provided attribute and value. */
+  const collectByAttribute = (attribute: string, value: string): HTMLElement[] => {
+    return Array.from(document.querySelectorAll<HTMLElement>(`[data-${attribute}='${value}']`)).filter(
+      node => !node.querySelector('.divider'),
```

Comment:
Since this function is performance-sensitive (it is called on every keystroke for each divider), we should avoid the nested querySelector.

I recommend adding a `.divider-thought` class to the `.thought` element in [StaticThought](https://github.com/cybersemics/em/blob/d1781ab3fc3db39a64ab15caba2e8625a5c3bead/src/components/StaticThought.tsx#L160). Then you can easily exclude the divider in the `querySelectorAll`.